src/Microsoft.SqlTools.ServiceLayer/AzureFunctions/AzureFunctionsUtils.cs

```diff
@@ -31,19 +34,71 @@ public static IEnumerable<MethodDeclarationSyntax> GetMethodsWithFunctionAttribu
+                .Parameters // Get all the parameters for the method
+                .SelectMany(p =>
+                    p.AttributeLists // Get a list of all attributes on any of the parameters
+                    .SelectMany(al => al.Attributes)
+                ).Where(a => a.Name.ToString().Equals(HTTP_TRIGGER_ATTRIBUTE_NAME) // Find any HttpTrigger attributes
+                ).FirstOrDefault() // Get the first one available - there should only ever be 0 or 1
+                ?.ArgumentList // Get all the arguments for the attribute
+                ?.Arguments
+                .Where(a => a.ChildNodes().OfType<NameEqualsSyntax>().Where(nes => nes.Name.ToString().Equals(ROUTE_ARGUMENT_NAME)).Any()) // Find the Route argument - it should always be a named argument
+                .FirstOrDefault()
+                ?.ChildNodes()
+                .OfType<ExpressionSyntax>() // Find the child identifier node with our value
+                .Where(le => le.Kind() != SyntaxKind.NullLiteralExpression) // Skip the null expressions so they aren't ToString()'d into "null"
+                .FirstOrDefault()
+                ?.ToString()
+                .TrimStart('$') // Remove $ from interpolated string, since this will always be outside the quotes we can just trim
+                .Trim('\"'); // Trim off the quotes from the string value - additional quotes at the beginning and end aren't valid syntax so it's fine to trim
+        }
+
+        /// <summary>
+        /// Gets the function name from the FunctionName attribute on a method
+        /// </summary>
+        /// <param name="m">The method</param>
+        /// <returns>The function name, or an empty string if the name attribute doesn't exist</returns>
+        public static string GetFunctionName(this MethodDeclarationSyntax m)
+        {
+            return m
+                .AttributeLists // Get all the attribute lists on the method
+                .Select(a =>
+                    a.Attributes.Where(
+                        attr => attr.Name.ToString().Equals(AzureFunctionsUtils.FUNCTION_NAME_ATTRIBUTE_NAME) // Find any FunctionName attributes
+                    ).FirstOrDefault() // Get the first one available - there should only ever be 0 or 1
+                ).Where(a => // Filter out any that didn't have a FunctionName attribute
+                    a != null
+                ).FirstOrDefault() // Get the first one available - there should only ever be 0 or 1
+                ?.ArgumentList // Get all the arguments for the attribute
+                ?.Arguments
+                .FirstOrDefault() // The first argument is the function name
+                ?.ToString()
+                .TrimStart('$') // Remove $ from interpolated string, since this will always be outside the quotes we can just trim
+                .Trim('\"') ?? ""; // Trim off the quotes from the string value - additional quotes at the beginning and end aren't valid syntax so it's fine to trim
```

Comment:
Why is this line different than 67 above?

Reply:
Because we expect routes to be null a lot of the time and that's valid, so we specifically return undefined in that case. But names should always have a value - even if that value is empty. Theoretically this should never happen, but the compiler doesn't know that so it complains that the value might be null at this point.